ledger/evalbench_test.go

```diff
@@ -97,27 +97,28 @@ func (g *BenchPaymentTxnGenerator) Txn(tb testing.TB, addrs []basics.Address, ke
 	return stxn
 }
 
-// BenchAppTxnGenerator generates app opt in transactions
-type BenchAppOptInsTxnGenerator struct {
+// benchAppOptInsTxnGenerator generates payment transactions acrosss accounts that have
+// opted into applications.
+type benchAppOptInsTxnGenerator struct {
 	NumApps             int
 	Proto               protocol.ConsensusVersion
 	Program             []byte
 	OptedInAccts        []basics.Address
 	OptedInAcctsIndices []int
+	MaxAppsOptedIn      int
+	TransactionsType    protocol.TxType
+	AppsOptedIn         map[basics.Address]map[basics.AppIndex]struct{}
 }
 
-func (g *BenchAppOptInsTxnGenerator) Prepare(tb testing.TB, addrs []basics.Address, keys []*crypto.SignatureSecrets, rnd basics.Round, gh crypto.Digest) ([]transactions.SignedTxn, int) {
+func (g *benchAppOptInsTxnGenerator) Prepare(tb testing.TB, addrs []basics.Address, keys []*crypto.SignatureSecrets, rnd basics.Round, gh crypto.Digest) ([]transactions.SignedTxn, int) {
 	maxLocalSchemaEntries := config.Consensus[g.Proto].MaxLocalSchemaEntries
-	maxAppsOptedIn := config.Consensus[g.Proto].MaxAppsOptedIn
-	if maxAppsOptedIn == 0 {
-		maxAppsOptedIn = config.Consensus[protocol.ConsensusV30].MaxAppsOptedIn
-	}
+	maxAppsOptedIn := g.MaxAppsOptedIn
 
-	// this function might create too much transaction even to fit into a single block
+	// this function might create more transactions than a single block could contain.
 	// estimate number of smaller blocks needed in order to set LastValid properly
```

Comment:
If that's the only reason for the extra work, why not just set LastValid to FirstValid + MaxTxLifetime?